app/models/factory_log.rb

```diff
@@ -16,29 +18,31 @@ def self.time_ranking
     ranking.sort_by!{|l| l[:total_time]}.reverse
   end
 
-  def self.average_time(factory)
-    sum_time(factory) / log_set_with_trait(factory).size
+  #Return specified factory's average exectuaion time.
+  def self.average_time(factory_name_and_traits)
+    logs = FactoryLog.log_set_with_trait(factory_name_and_traits)
+    logs.map{ |log| log.time }.inject{ |sum, n| sum + n } / logs.size
```

Comment:
Space missing to the left of {.<br>Pass `&:time` as an argument to `map` instead of a block.